utils/session/with-session.ts

```diff
@@ -0,0 +1,20 @@
+import { IronSession, getIronSession } from 'iron-session';
+import { NextApiRequest, NextApiResponse } from 'next';
+import { ISession, sessionOptions } from '.';
+
+export type IReqWithSession = NextApiRequest & {
+  session: IronSession<ISession>;
+};
+export default function withSession(
```

Comment:
`withIronSessionApiRoute` is not exported anymore with the last version of `iron-session`. I made a custom high order function with stricter typing.